### ISSUE TITLE
Why does `.holdWhen(valve)` swallow events from finite streams (regardless of `valve`)?

### ISSUE THREAD

**Issue by timmolendijk** [baconjs/bacon.js#542]
I don't understand why the following contrast exists:

```javascript
Bacon.once(true).log("you will see me once");
// and then it will log `<end>`
```

versus

```javascript
Bacon.once(true).holdWhen(Bacon.constant(false)).log("you won't ever see me");
// it will just log `<end>`
```

This odd behavior disappears when `Bacon.once(true)` is replaced with `Bacon.interval(1000, true)` (or a similar non-completing stream).

I've tried to [dig in the code](https://github.com/baconjs/bacon.js/blob/master/src/holdwhen.coffee) but I'm having a hard time understanding what's going on there.

**Comment by phadej** [comment]
(In your example it probably should be `Bacon.constant(true)`)

It might be related to
```js
Bacon.constant(true).log("t")
// t true
// t <end> // <- breaks `holdWhen` ?
```

Because

```js
var b = new Bacon.Bus();
b.log();
var p = b.toProperty(true);
p.log("t")
// t true <- no <end>!
Bacon.once(1).holdWhen(p).log("you won't see me");
// nothing printed
```

**Comment by raimohanska** [comment]
@phadej in your example, `holdWhen` works correctly, because the property holds the value `true`. But if you

```js
Bacon=require("../dist/Bacon.js")
var b = new Bacon.Bus()
var p = b.toProperty(false);
Bacon.once(1).holdWhen(p).log("you won't see me");
// nothing printed
```

you can indeed reproduce the bug.

**Comment by raimohanska** [comment]
If you replace the bus with `Bacon.never()` the bug disappears, and this probably has to do with the fact that the property has ended.

**Comment by raimohanska** [comment]
Seems that `holdWhen` is a particularly hard thing to get right :(

**Comment by raimohanska** [comment]
If you replace `Bacon.once(1)` with `Bacon.later(0,1)` it works fine. So this is one of the things that breaks because of synchronously responding streams.

**Comment by raimohanska** [comment]
Related to #367 the fix for which is on my wish-list for version 0.8.

**Comment by raimohanska** [comment]
There's not much point in using `.holdWhen` for a synchrnous stream like `Bacon.once()`, right? I don't think we'll fix this one until #367 gets fixed, if ever.
